common/src/main/java/com/sequenceiq/cloudbreak/telemetry/DataBusEndpointProvider.java

```diff
@@ -39,25 +48,49 @@ public String getDataBusEndpoint(String endpoint, boolean useCname) {
         return databusUrl;
     }
 
-    public String getDatabusS3Endpoint(String endpoint, boolean forceToUsePatterns) {
+    public Optional<String> getDatabusS3Endpoint(String endpoint, String region) {
+        return getDatabusS3Endpoint(endpoint, false, region);
+    }
+
+    public Optional<String> getDatabusS3Endpoint(String endpoint, boolean forceToUsePatterns, String region) {
+        if (fipsEnabled && region.contains(GOV_REGION_INDICATOR)) {
```

Comment:
can we somehow the pass the isgovcloud flag from the credential ?

Reply:
credential is not accessible from stack, i would have to call env service to find out, just like here: https://github.com/hortonworks/cloudbreak/blob/master/core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/EnvironmentServiceDecorator.java#L71-L74 , i would not make an expensive call here, since it is used in converter and decorator classes

Reply:
also region is needed anyway to construct the region dependent fips compliant endpoints

Reply:
what i can do here is an extra check based on region in resource crn

Reply:
is this resolved @doktoric ? can this be merged?